lib/splitclient-rb/cache/senders/impressions_sender.rb

```diff
@@ -31,13 +31,19 @@ def call
         private
 
         def impressions_thread
-          Thread.new do
-            @config.logger.info('Starting impressions service')
+          @config.threads[:impressions_sender] = Thread.new do
+            begin
+              @config.logger.info('Starting impressions service')
 
-            loop do
+              loop do
+                post_impressions
+
+                sleep(SplitIoClient::Utilities.randomize_interval(@config.impressions_refresh_rate))
```

Comment:
@maksimf remember that this cause an infinite loop before. If the post errors out, it skips the sleep and cycles again there is a nextwork issue. Please restablish the previous pattern.